torch/_functorch/_aot_autograd/traced_function_transforms.py

```diff
@@ -470,6 +485,10 @@ def joint_helper(primals, tangents):
         # Setup the wrapper for functionalization of rng ops
         helper, args = create_functionalized_rng_ops_wrapper(helper, args, trace_joint)
 
+    # Additionally pass in tokens as inputs
+    additional_token_inputs = [torch.tensor([])] * len(meta.tokens)
+    args = [*additional_token_inputs, *args]
+
```

Comment:
What happens in the requires_grad=True case? We trace out a joint fwd_backward graph that has tokens, and then partition it? If so, can we add an assertion that all the tokens show up in the partitioned fwd graph?

Reply:
related question: what happens if someone wants to use an op that requires an effect token in the backward (e.g. in a custom autograd.Function)?

I think we should either:
(1) explicitly support it in this PR (doable but kind of a pain, since the backward doesn't have a runtime wrapper today, and we also don't do any "analysis" on the backward to see if there are any ops that needs this handling)
(2) figure out how to explicitly error and wait for someone to run into it

Reply:
When requires_Grad=True, right now I just mark the tokens as "forward outputs" that need to be partitioned in the forward graph and not the backwards graph. There are a couple of assertions checking that the number of input/output nodes of the graphs are consistent with the ViewAndMutationMetadata, so I think those would be the "explicit errors" one might see if there ends up being a token in the backwards graph?

Reply:
seems reasonable